src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.cs

```diff
@@ -174,7 +174,18 @@ private void ReleaseConnection(SqlConnection connection)
                     return;
                 }
 
-                _connectionsPool.Push(connection);
+                try
+                {
+                    _connectionsPool.Push(connection);
```

Comment:
Is it guaranteed that Push() throwing an OutOfMemoryException leaves the _connectionsPool in a functional state?

Reply:
The only thing we're relying on here is that `_connectionsPool` doesn't contain `connection` if an `OutOfMemoryException` is thrown.

Reply:
This seems like a safe change since the exception is getting rethrown.